src/util/json.h

```diff
@@ -169,13 +171,63 @@ class json_arrayt:public jsont
     array.push_back(jsont());
     return array.back();
   }
+
+  template <typename... argumentst>
+  void emplace_back(argumentst &&... arguments)
+  {
+    array.emplace_back(std::forward<argumentst>(arguments)...);
+  }
+
+  std::vector<jsont>::iterator begin()
+  {
+    return array.begin();
+  }
+
+  std::vector<jsont>::const_iterator begin() const
+  {
+    return array.begin();
+  }
+
+  std::vector<jsont>::const_iterator cbegin() const
+  {
+    return array.cbegin();
+  }
+
+  std::vector<jsont>::iterator end()
+  {
+    return array.end();
+  }
+
+  std::vector<jsont>::const_iterator end() const
+  {
+    return array.end();
+  }
+
+  std::vector<jsont>::const_iterator cend() const
+  {
+    return array.cend();
+  }
+
+  typedef jsont value_type; // NOLINT(readability/identifiers)
```

Comment:
What's the point of this?

Reply:
To make more complaint with std algorithms

Reply:
It might help to add some concrete examples in the commit message.

Reply:
I discussed this with @thk123 in person. He hadn't actually read the existing message on this commit. I will however come up with a simple example to add to the commit message.

Reply:
@tautschnig Added.